persistence/sql/persister_identity.go

```diff
@@ -472,3 +510,65 @@ func (p *Persister) injectTraitsSchemaURL(ctx context.Context, i *identity.Ident
 	i.SchemaURL = s.SchemaURL(p.r.Config(ctx).SelfPublicURL(nil)).String()
 	return nil
 }
+
+func (p *Persister) getJsonSearchQuery(field string, values []string) pop.ScopeFunc {
+	return func(q *pop.Query) *pop.Query {
+		dsn := p.r.Config(context.Background()).DSN()
+		switch {
+		case strings.HasPrefix(dsn, "sqlite"), strings.HasPrefix(dsn, "mysql"):
+			field, innerField := extractFieldAndInnerFields(field)
+			for _, value := range values {
+				if innerField == "" {
+					q = q.Where(fmt.Sprintf(`json_extract(%s, '$') = "%v"`, field, value))
+				} else {
+					q = q.Where(fmt.Sprintf(`json_extract(%s, '$.%v') = "%v"`, field, innerField, value))
+				}
+			}
+			return q
+		case strings.HasPrefix(dsn, "postgres"), strings.HasPrefix(dsn, "crdb"), strings.HasPrefix(dsn, "cockroach"):
+			field, innerField := extractFieldAndInnerFields(field)
+			for _, value := range values {
+				if innerField == "" {
+					q = q.Where(fmt.Sprintf(`%s @> '"%s"'`, field, value))
+				} else {
+					q = q.Where(fmt.Sprintf(`%s @> '{"%s":"%s"}'`, field, innerField, value))
+				}
+			}
+			return q
+		}
+		return q
+	}
+}
+
+func (p *Persister) buildScope(values url.Values) pop.ScopeFunc {
+	return func(q *pop.Query) *pop.Query {
+		for field, value := range values {
+			if IsStringInSlice([]string{"page","per_page"}, field) {
+				continue
+			}
+			if strings.HasPrefix(field, "traits") {
+				q = q.Scope(p.getJsonSearchQuery(field, value))
+				continue
+			}
+			q = q.Where(fmt.Sprintf("%s IN (?)", field), value)
```

Comment:
Here also `field` could be used for SQL injection.

Reply:
The point is I tried to use sql parameter here but it didn't work because of value that is an array